src/resources/env_config.rs

```diff
@@ -0,0 +1,233 @@
+//! Configuration options that can be passed through environment variables.
+//!
+//! By default, mdBook will try to retrieve external resources through a content
+//! delivery network. If that fails, it'll load an embedded version of the
+//! resource. Using environment variables this behavior can be changed in
+//! various ways.
+//!
+//! The variables are as follows. [name] can be one of JQUERY, MATHJAX,
+//! HIGHLIGHT, AWESOME, OPEN_SANS, SOURCE_CODE_PRO:
+//!
+//! - MDBOOK_GLOBAL_STRATEGY: Strategy for finding external resources. One of
+//!   UrlWithFallback (the default), UrlOnly and Omit.
+//! - MDBOOK_[name]_URL: The URL of the resource.
+//! - MDBOOK_[name]_SOURCE: The source file for the embedded resource.
+//! - MDBOOK_[name]_STRATEGY: Strategy for this particular resource. This
+//!   overrides the global strategy.
+//!
+//! All variables have sane defaults which were determined at compile time.
+//! the SOURCE variables, when left empty, ensure that a version is used that
+//! was embedded into mdBook itself at compile time.
+
+use std::env;
+
+pub static JQUERY_URL: &'static str = &"https://code.jquery.com/jquery-2.1.4.min.js";
+pub static MATHJAX_URL: &'static str = &"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
+pub static HIGHLIGHT_URL: &'static str = &"highlight.js";
+pub static AWESOME_URL: &'static str = &"https://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css";
+pub static OPEN_SANS_URL: &'static str = &"https://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800";
+pub static SOURCE_CODE_PRO_URL: &'static str = &"https://fonts.googleapis.com/css?family=Source+Code+Pro:500";
+
+/// A third-party resource
+pub trait Resource {
+    /// returns true if this resource has to be rendered as an url
+    fn must_render_url(&self) -> bool;
+    /// returns true if this resource has to be embedded in the book
+    fn must_embed(&self) -> bool;
+    /// returns the url for this resource. This panics if must_render_url returns false.
+    fn url(&self) -> String;
+    /// returns the source location for this resource, or None if none was configured.
+    /// This panics if must_embed returns false.
+    fn source(&self) -> Option<String>;
+}
+
+/// A third-party resource with nothing special to it.
+pub enum BasicResource {
```

Comment:
Any chance you can make this guy derive `Debug` and `Clone`? Not having a debug impl makes things annoying when you are debugging.